### PR TITLE
#830 Inbound and Outbound Shipment mutations

### LINKED ISSUE #830: Shipments API issues
Part of #825 - the API for invoices is broken with the latest changes

- [x] List view for inbound and outbound
- [x] Detail view for inbound and outbound
- [x] Inbound mutations
- [x] Outbound mutations
- [x] Dashboard request to invoices

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #830 

Fixes the mutations for invoices and updates the colour picker. put in a fragment for the Outbound shipment list and then decided that this PR was already too big.. so haven't done the other three

### CODE REVIEW

File: packages/invoices/src/InboundShipment/ListView/ListView.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ export const InboundListView: FC = () => {
   const navigate = useNavigate();
   const { error } = useNotification();
   const api = useInboundShipmentApi();
+  const { storeId } = useQueryParams({ initialSortBy: { key: 'id' } });
```
Comment (mark-prins) [inline packages/invoices/src/InboundShipment/ListView/ListView.tsx:28]: need to do something different.. ok for now in order to get things going..?

File: packages/invoices/src/OutboundShipment/ListView/ListView.tsx [hunk 2]
```diff
@@ -63,7 +65,10 @@ export const OutboundShipmentListViewComponent: FC = () => {
       'comment',
       [
         'totalAfterTax',
-        { accessor: ({ rowData }) => rowData.pricing.totalAfterTax },
+        {
+          accessor: ({ rowData }) =>
+            pricingGuard(rowData.pricing).totalAfterTax,
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/ListView/ListView.tsx:70]: Nice.. I'm hopeful the changes will come through for invoices soon also

**Review by joshxg — APPROVED** [review]
Very nice! I pushed a merge of main and some test cases because I'm slow and took a bit to understand the bug you had found (but thanks for fixing it 😁 )

File: packages/invoices/src/OutboundShipment/DetailView/modals/hooks/usePackSizeController.tsx [hunk 3]
```diff
@@ -74,12 +74,20 @@ export const usePackSizeController = (lines: DraftOutboundLine[]) => {
...
       -1
     );
 
+    // on initial selection of an item there will be no packs selected
+    if (selectedPackSize === -1)
+      selectedPackSize = ifTheSameElseDefault(
+        lines.filter(({ id }) => id !== 'placeholder'),
+        'packSize',
+        -1
+      );
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/DetailView/modals/hooks/usePackSizeController.tsx:92]: ah gosh! thanks - forgot about adding tests

**Pull request merged**